packages/hardhat/test/integration-tests/periphery/controller-helper.ts

```diff
@@ -1760,5 +1885,436 @@ describe("Controller helper integration test", function () {
+  
+        const params = {
+          wPowerPerpPool: wSqueethPool.address,
+          tokenId: oldTokenId,
+          ethAmountToLp: BigNumber.from(0),
+          liquidity: oldPosition.liquidity,
+          wPowerPerpAmountDesired: ethers.utils.parseUnits('0'),
+          wethAmountDesired: wethAmountInLp,
+          amount0DesiredMin: BigNumber.from(0),
+          amount1DesiredMin: BigNumber.from(0),
+          limitPriceEthPerPowerPerp: BigNumber.from(0),
+          amount0Min: BigNumber.from(0),
+          amount1Min: BigNumber.from(0),
+          lowerTick: isWethToken0 ? newTick: -887220,
+          upperTick: isWethToken0 ? 887220: newTick,
+          poolFee: 3000
+        }
+  
+        await (positionManager as INonfungiblePositionManager).connect(depositor).approve(controllerHelper.address, oldTokenId);
+        // console.log('before rebalanceWithoutVault')
+        await controllerHelper.connect(depositor).rebalanceWithoutVault(params);
+        // console.log('after rebalanceWithoutVault')
+        const depositorEthBalanceAfter = await provider.getBalance(depositor.address)
+
+
+        tokenIndexAfter = await (positionManager as INonfungiblePositionManager).totalSupply();
+        const newTokenId = await (positionManager as INonfungiblePositionManager).tokenByIndex(tokenIndexAfter.sub(1));
+        const newPosition = await (positionManager as INonfungiblePositionManager).positions(newTokenId);
+        const ownerOfUniNFT = await (positionManager as INonfungiblePositionManager).ownerOf(newTokenId); 
+  
+        const [amount0New, amount1New] = await (positionManager as INonfungiblePositionManager).connect(depositor).callStatic.decreaseLiquidity({
+          tokenId: newTokenId,
+          liquidity: newPosition.liquidity,
+          amount0Min: BigNumber.from(0),
+          amount1Min: BigNumber.from(0),
+          deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+        })
+  
+        const wPowerPerpAmountInNewLp = (isWethToken0) ? amount1New : amount0New;
+        const wethAmountInNewLp = (isWethToken0) ? amount0New : amount1New;
+        // console.log('wPowerPerpAmountInLP',wPowerPerpAmountInLp.toString())
+        // console.log('wethAmountInLp', wethAmountInLp.toString())
+        // console.log('wPowerPerpAmountInNewLp',wPowerPerpAmountInNewLp.toString())
+        // console.log('wethAmountInNewLp', wethAmountInNewLp.toString())
+        // console.log('depositor address', depositor.address);
+        // console.log('depositorEthBalanceBefore', depositorEthBalanceBefore.toString());
+        // console.log('depositorEthBalanceAfter', depositorEthBalanceAfter.toString());
+        expect(ownerOfUniNFT === depositor.address).to.be.true;
+        // Sold all wPowerPerp in LP
+        expect(wPowerPerpAmountInNewLp.lte(10)).to.be.true
+        // weth stays the same
+        expect((wethAmountInNewLp.sub(wethAmountInLp)).abs().lte(10)).to.be.true
```

Comment:
Should check if the oSQTH is sold and if the user gets the ETH amount from the sale.